### ISSUE TITLE
aria-valid-attr-value WCAG mapping

### ISSUE THREAD

**Issue by lilzesh** [dequelabs/axe-core#4902]
### Product

axe-core

### Question

I was testing some pages recently which made me wonder about the [aria-valid-attr-value rule](https://github.com/dequelabs/axe-core/blob/b91369a1690a9d0a8b3e580823efdcd7878d22ec/lib/rules/aria-valid-attr-value.json) and why it maps exclusively to 4.1.2. I understand that ARIA attributes like aria-label and aria-labelledby certainly have direct 4.1.2 implications but there are other attributes handled by the [check of this rule](https://github.com/dequelabs/axe-core/blob/b91369a1690a9d0a8b3e580823efdcd7878d22ec/lib/checks/aria/aria-valid-attr-value-evaluate.js) that do not seem to fit with the 4.1.2 tagging (e.g. aria-controls, aria-describedby, among others). Invalid values for some of these other attributes, to me, would align more appropriately with 1.3.1 as it is likely that we would be dealing with some sort of information or relationship expressed visually that cannot be programmatically determined. With some of these other attributes we wouldn't be dealing with the programmatic determinability of name and role, and some are not settable by a user.

I wonder if it makes sense to separate this rule and check into attributes whose invalid values are 4.1.2 violations and others that are 1.3.1 violations? Both likely still make sense tagged as needs review. Also, if this type of discussion has occurred in the past and what may be limiting factors of my argument as I acknowledge I'm lacking a lot of historical background.

**Comment by lilzesh** [comment]
I do find the [defintion of name](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html#dfn-name), in the context of the normative text of 4.1.2 curious. To me this implies that name is actually a superset that contains not only accessible name but any text that helps identify a component, which very well could be a description. Perhaps this is a reason something like `aria-describedby` might sneak into 4.1.2.

**Comment by WilcoFiers** [comment]
That's a good question. Most attributes that can fail `aria-valid-attr-value` are related to 4.1.2. aria-describedby for example can never fail it. It's okay for that to be empty, it's okay for that to have non-existing IDs... axe cannot confidently fail anything about that. The same is true for other attributes that use IDs (DREFS properties).

The scope of 4.1.2 isn't attribute related, it's related to what type of element its on. An invalid ARIA property on a button always falls under 4.1.2. We could in theory split all of those rules up to have elements on interactive roles mapped to 4.1.2 and elements on non-interactive roles to 1.3.1. That's gets us a little closer to a "correct" mapping. But it isn't going to be perfect either. For separators you can't just look at the role. In interactive separator is different from a static one for instance. Elements may be interactive without having the correct role, so maybe you include those? But then you end up with too many elements, so maybe you look at other properties to try and guess the intended role. Etc.  We can endlessly tweak an algorithm like that and never get it right. 

That's why generally we opt for keeping things predictable. We did a similar thing with impact. Until fairly recently axe would sometimes report the issue of a rule as critical, and other times as serious, depending on the situation. That kind of worked, but its far from accurate. We made a best effort based on what the architecture of axe allows us to. But people kept asking why its sometimes critical and other times serious, and assuming axe did something much smarter than what it actually does. The little bit this made axe better wasn't worth the extra confusion this created for people trying to understand the results.
